adoc/chapters/programming_interface.adoc

```diff
@@ -14622,6 +14622,13 @@ interoperability member functions provided by the [code]#interop_handle# class.
 
 Local <<accessor,accessors>> cannot be used within a <<host-task>>.
 
+If a C++ lambda is passed to a <<host-task>>, the lambda may capture by
+reference or by value. If a <<host-task>> lambda captures by reference, the user
+must ensure that referred-to user defined classes use appropriate reference
+counting. Since the <<host_task>> callable may execute at an unspecified time
+between a command submission and a synchronization point, its execution may
+outlive the referred-to objects if reference counting is not used.
```

Comment:
This seems too specific to a particular bug that you had, and the recommendation about reference counting is only one possible solution.  I don't think we need to add any new formal requirements to the spec here.  We already say that a host task's execution is deferred and that it executes asynchronously.  A logical outcome of this is that you need to take care when using pointers or references inside the host task.

If we want to add something to the spec, I'd suggest a non-normative note like this:

```
[NOTE]
====
Care must be taken when using pointers or references inside a host task when the pointer
or reference refers to an object that is defined outside of the host task.  Because the host
task is executed asynchronously, applications should ensure that the referenced object's
lifetime lasts at least until the host task completes.
====
```

Reply:
Thanks for comments. Have rephrased to advise caution